tx-pool/src/component/container.rs

```diff
@@ -401,18 +442,10 @@ impl SortedTxMap {
     // we are sure that all in-pool ancestor have already been processed.
     // otherwise `links` will differ from the set of parents we'd calculate by searching
     pub fn remove_entry(&mut self, id: &ProposalShortId) -> Option<TxEntry> {
-        let descendants = self.calc_descendants(id);
         self.remove_unchecked(id).map(|entry| {
             // We're not recursively removing a tx and all its descendants
             // So we need update statistics state
-            for desc_id in &descendants {
-                if let Some(desc_entry) = self.entries.get_mut(desc_id) {
-                    let deleted = self.sorted_index.remove(&desc_entry.as_sorted_key());
-                    debug_assert!(deleted, "pool inconsistent");
-                    desc_entry.sub_entry_weight(&entry);
-                    self.sorted_index.insert(desc_entry.as_sorted_key());
-                }
-            }
+            self.update_descendants_index_key(&entry, EntryOp::Remove);
```

Comment:
How about change `EntryOp::Remove` to `EntryOp::Sub`, since its purpose isn't "remove something" but "descrease weight".